Jenkinsfile

```diff
@@ -0,0 +1,153 @@
+        storage:
+          strategy: ephemeral
+    metrics:
+      prometheus:
+        enabled: true
+        storage:
+          strategy: ephemeral
+  transports:
+    qdr:
+      enabled: true
+      deployment_size: 1
+      web:
+        enabled: false
+  elasticsearch_manifest: |
+    apiVersion: elasticsearch.k8s.elastic.co/v1beta1
+    kind: Elasticsearch
+    metadata:
+      name: elasticsearch
+      namespace: $namespace
+    spec:
+      version: 7.10.2
+      http:
+        tls:
+          certificate:
+            secretName: 'elasticsearch-es-cert'
+      nodeSets:
+        - config:
+            node.data: true
+            node.ingest: true
+            node.master: true
+            node.store.allow_mmap: true
+          count: 1
+          name: default
+          podTemplate:
+            metadata:
+              labels:
+                tuned.openshift.io/elasticsearch: elasticsearch
+            spec:
+              containers:
+                - name: elasticsearch
+                  resources:
+                    limits:
+                      cpu: '2'
+                      memory: 4Gi
+                    requests:
+                      cpu: '1'
+                      memory: 2Gi
+              volumes:
+                - emptyDir: {}
+                  name: elasticsearch-data
+"""
+
+podTemplate(containers: [
+    containerTemplate(
+        cloud: 'openshift',
+        name: 'exec',
+        image: 'image-registry.openshift-image-registry.svc:5000/ci/jenkins-agent:latest',
+        command: 'sleep',
+        args: 'infinity',
+        alwaysPullImage: true
+    )],
+    serviceAccount: 'jenkins-operator-cloudops'
+
+) {
+    node(POD_LABEL) {
+        container('exec') {
+            dir('service-telemetry-operator') {
+                stage ('Clone Upstream') {
+                    catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                        checkout scm
+                    }
+                }
+                stage ('Create project') {
+                    if ( currentBuild.result == 'FAILURE' ) { stages_failed = true; return; }
+                    catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                        openshift.withCluster(){
+                            openshift.newProject(namespace)
+                        }
+                    }
+                }
+                stage('Build STF Containers') {
+                    if ( currentBuild.result == 'FAILURE' ) { stages_failed = true; return; }
+                    catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                        ansiColor('xterm') {
+                            ansiblePlaybook(
+                                playbook: 'build/run-ci.yaml',
+                                colorized: true,
+                                extraVars: [
+                                    "namespace": namespace,
+                                    "__deploy_stf": "false",
```

Comment:
Huh this works? I'd think you'd need STF deployed in order for the validation and smoke tests to pass?

Reply:
STF is deployed in the next step of the Jenkins file. I needed to use a custom manifest to minimize the ES footprint to get reliable deployments on Quicklab, so I deploy it straight from Jenkinsfile

Reply:
So this call to run-ci is actually more like build-containers-for-ci? I kinda get that from the stage name, but a one-line comment above L112 saying something like "Builds containers but does not actually run CI" might be helpful